### PR TITLE
Updated Gradle to 7.4.2

### PR DESCRIPTION
**Pull request by marine-master**
Changes:

- The Gradle wrapper has been upgraded to version 7.4.2.
- build.gradle has been fully updated for Gradle 7. There are no warnings.
- Gradle's own dependency verification mechanism has taken the place of "gradle-witness".
- Task "test" and "build" are now working under Java 17.
- Additional comments have been added.

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+         <artifact name="jna-4.5.2.jar">
+            <sha512 value="41d3b175af8aca64dced867f137149019d3a2c0e45eacb84c3a7d066b1fcb66de154c0d0739835df22bf6609f7e779e017878dd27452c59213368cf73fb83f2d" origin="Generated by Gradle"/>
+         </artifact>
+         <artifact name="jna-4.5.2.pom">
+            <sha512 value="7221025ec583a451318883d349c1e670e4f70332f1c871caece8843e267b185b426ee9905dc3c0146ef47c2c5256f5321b0885354b5501aa49d6f8309268a51d" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
+      <component group="net.java.dev.jna" name="jna-platform" version="4.5.2">
+         <artifact name="jna-platform-4.5.2.jar">
+            <sha512 value="440e81a7c7ded403c3190d81dc372a6559ffc4f9cb13d7d07328f25c34887f2df594652aeb04308ded8b473a5422ef45c58c7d6d2677ce541372fb52afc7a792" origin="Generated by Gradle"/>
```
Comment (ArneBab) [inline gradle/verification-metadata.xml:51]: This cannot be matched with the dependencies.properties, which is the actual source of truth.

Can Gradle include the sha256 hash instead?

dependencies.properties includes this about jna-platform:

```
jna-platform.filename=jna-platform-4.5.2.jar
jna-platform.sha256=f1d00c167d8921c6e23c626ef9f1c3ae0be473c95c68ffa012bc7ae55a87e2d6
```

We need an entry that connects the gradle-hash to the dependencies.properties hash to be able to switch.

**Review by ArneBab — CHANGES_REQUESTED** [review]

**Comment by ArneBab** [comment]
Thank you for the work on updating gradle! I’m sorry that I didn’t answer earlier: I had missed your PR.

**Comment by hernic** [comment]
@ArneBab @Bombe hello, it could be nice for Virtual Threads if this PR could be resolved and merged (needed)

**Comment by ArneBab** [comment]
Closing in favor of #827 which already uses sha256 so its validation can be checked against dependencies.properties.

@hernic I hope #827 will move forward.

**Pull request closed without merging**